tiledb/sm/enums/datatype.h

```diff
@@ -347,6 +352,31 @@ inline bool datatype_is_time(Datatype type) {
       type == Datatype::TIME_AS);
 }
 
+inline uint8_t datatype_to_underlying(Datatype type) {
+  return static_cast<uint8_t>(type);
```

Comment:
> `uint8_t`

It might be worth defining an alias for this to use here and in the definition of enum class. Something like `using datatype_underlying_t = uint8_t;`. Using the alias expresses exactly what the purpose of the cast is without as much low-information repetition in comments.